### ISSUE TITLE
understadning output

### ISSUE THREAD

**Issue by GraninV** [OlafenwaMoses/ImageAI#550]
I will be happy for a small explanation.
Running ResNet. I can't understand why there are two sets of data under each epoch. What does it mean 10/10 and 111/111. Why model saving coming before finishing this epoch? There is totally no information in the manual.
Thank you.

```
Using Enhanced Data Generation
Found 442 images belonging to 2 classes.
Found 38 images belonging to 2 classes.
JSON Mapping for the model classes saved to  /content/drive/My Drive/Colab Notebooks/images/ML/json/model_class.json
Number of experiments (Epochs) :  100


Epoch 1/100
10/10 [==============================] - 22s 2s/step - loss: 7.6561 - acc: 0.5000
Epoch 00001: val_acc improved from -inf to 0.50000, saving model to /content/drive/My Drive/Colab Notebooks/images/ML/models/model_ex-001_acc-0.500000.h5
111/111 [==============================] - 297s 3s/step - loss: 0.7466 - acc: 0.7941 - val_loss: 7.6561 - val_acc: 0.5000

Epoch 2/100
10/10 [==============================] - 0s 28ms/step - loss: 7.6561 - acc: 0.5000
Epoch 00002: val_acc did not improve from 0.50000
111/111 [==============================] - 12s 105ms/step - loss: 0.3910 - acc: 0.8778 - val_loss: 7.6561 - val_acc: 0.5000
```

**Comment by AbimbolaOO** [comment]
Hello @GraninV,

10/10 means that for a single epoch the training data was split by the batch size into 10 batches so the loop going over all of the datasets has to run 10 times to cover all of the data for a single epoch. This Is repeat for 100 epoch form the what you out log is showing. 

The model is saving before the epoch finishes are just a way to simulate early stopping. What that means is that while your models are still training for the numbers of epoch you have specified, you want it to save any time the model generalization error or loss is decreasing or anytime you test accuracy increases and these can happen anytime during training so you want to save the model anytime it happens.
